pkg/runtime/scheduler/scheduler.go

```diff
@@ -136,147 +154,218 @@ func (m *Manager) processStream(ctx context.Context, client *client.Client, stre
-	parts := strings.Split(resp.GetName(), "||")
-	jobName := parts[len(parts)-1]
-
-	switch m.isHTTP {
-	case true:
-		req := v1.NewInvokeMethodRequest("dapr/receiveJobs/"+jobName).
-			WithHTTPExtension(http.MethodPost, "").
-			WithDataObject(resp.GetData())
-		if req != nil {
-			defer req.Close()
+func (m *Manager) processTriggers(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case job := <-m.jobCh:
+			log.Debugf("Received scheduled Job")
+			m.wg.Add(1)
+			go func() {
+				defer m.wg.Done()
+				m.handleJob(ctx, job)
+			}()
 		}
+	}
 
-		if m.appChannel == nil {
-			// TODO(Cassie): add an orphaned job go routine to retry sending job at a later time
-			return fmt.Errorf("app channel is nil. Cannot send back triggered job: %s", jobName)
-		}
+	return nil
+}
 
-		response, err := m.appChannel.TriggerJob(ctx, req)
-		if response != nil {
-			defer response.Close()
-		}
-		if err != nil {
-			// TODO(Cassie): add an orphaned job go routine to retry sending job at a later time
-			return fmt.Errorf("error returned from app channel while sending triggered job to app: %w", err)
+func (m *Manager) handleJob(ctx context.Context, job *schedulerv1pb.WatchJobsResponse) {
+	meta := job.GetMetadata()
+
+	switch t := meta.GetType(); t.GetSource().(type) {
+	case *schedulerv1pb.ScheduleJobMetadataType_App:
+		if err := m.invokeApp(ctx, job); err != nil {
+			log.Errorf("failed to invoke schedule app job: %s", err)
 		}
 
-		statusCode := int(response.Status().GetCode())
-		switch {
-		case statusCode >= 200 && statusCode <= 299:
-			log.Debugf("Sent job: %s to app: %s", jobName, m.appID)
-		case statusCode == http.StatusNotFound:
-			body, _ := response.RawDataFull()
-			log.Errorf("non-retriable error returned from app while processing triggered job %v: %s. status code returned: %v", jobName, body, statusCode)
-			return nil
-		default:
-			log.Errorf("unexpected status code returned from app while processing triggered job %s. status code returned: %v", jobName, statusCode)
+	case *schedulerv1pb.ScheduleJobMetadataType_Actor:
+		if err := m.invokeActorReminder(ctx, job); err != nil {
+			log.Errorf("failed to invoke scheduled actor reminder: %s", err)
 		}
+
 	default:
-		req := v1.NewInvokeMethodRequest("receiveJobs/" + jobName).
-			WithDataObject(resp.GetData())
-		if req != nil {
-			defer req.Close()
-		}
+		log.Errorf("Unknown job metadata type: %+v", t)
+	}
+}
 
-		if m.appChannel == nil {
-			// TODO(Cassie): add an orphaned job go routine to retry sending job at a later time
-			return fmt.Errorf("app channel is nil. Cannot send back triggered job: %s", jobName)
-		}
+func (m *Manager) invokeApp(ctx context.Context, job *schedulerv1pb.WatchJobsResponse) error {
+	appChannel := m.channels.AppChannel()
+	if appChannel == nil {
+		return errors.New("received app reminder but app channel not initialized")
```

Comment:
Do we know its a reminder type job here or could it also be the job 'type'?
```suggestion
		return errors.New("received triggered job, but app channel not initialized")
```